File: leetcode-js/365-water-and-jug-problem.js
const gcd = require('./util/gcd');

/**
 * @param {number} x
 * @param {number} y
 * @param {number} z
 * @return {boolean}
 */
var canMeasureWater = function(x, y, z) {
    if (x + y < z) {
        return false;
    }
    if (x === z || y === z || x + y === z) {
        return true;
    }
    return z % gcd(x, y) === 0;
};

File: leetcode-java/365-water-and-jug-problem.java
public class Solution {
    public boolean canMeasureWater(int x, int y, int z) {
        if (x + y < z) {
            return false;
        }

        if (x == z || y == z || x + y == z) {
            return true;
        }

        return z % gcd(x, y) == 0;
    }

    int gcd(int a, int b) {
        while (b != 0) {
            int t = b;
            b = a % b;
            a = t;
        }
        return a;
    }
}
